include/flucoma/clients/nrt/KDTreeClient.hpp

```diff
@@ -105,19 +97,9 @@ class KDTreeClient : public FluidBaseClient,
   MessageResult<RealVector> kNearestDist(InputBufferPtr  data,
                                          Optional<index> nNeighbours) const
   {
-    // TODO: refactor with kNearest
     index k = nNeighbours ? nNeighbours.value() : get<kNumNeighbors>();
-    if (k > mAlgorithm.size()) return Error<RealVector>(SmallDataSet);
-    // if (k <= 0 && get<kRadius>() <= 0) return Error<RealVector>(SmallK);
-    if (!mAlgorithm.initialized()) return Error<RealVector>(NoDataFitted);
-    InBufferCheck bufCheck(mAlgorithm.dims());
-    if (!bufCheck.checkInputs(data.get()))
-      return Error<RealVector>(bufCheck.error());
-    RealVector point(mAlgorithm.dims());
-    point <<=
-        BufferAdaptor::ReadAccess(data.get()).samps(0, mAlgorithm.dims(), 0);
-    auto [dist, ids] = mAlgorithm.kNearest(point, k, get<kRadius>());
-    return {dist};
+    auto  reply = computeKnearest(data, k);
```

Comment:
You need to check that the result is `ok` here as well

Reply:
oupsy, sorry, added now running the tests